crates/runtime/src/virtual_machine.rs

```diff
@@ -326,79 +337,100 @@ impl VirtualMachine {
+                    // selection - the options handler must have
+                    // called SetSelectedOption! Continue running
+                    // immediately.
+                    *self.execution_state_mut() = ExecutionState::Running;
+                }
+                self.state_mut().program_counter += 1;
             }
             OpCode::PushString => {
                 //Pushes a string value onto the stack. The operand is an index into the string table, so that's looked up first.
                 let string_table_index: String = instruction.read_operand(0);
-                self.state.push(string_table_index);
+                self.state_mut().push(string_table_index);
+                self.state_mut().program_counter += 1;
             }
             OpCode::PushFloat => {
                 // Pushes a floating point onto the stack.
                 let float: f32 = instruction.read_operand(0);
-                self.state.push(float);
+                self.state_mut().push(float);
+                self.state_mut().program_counter += 1;
             }
             OpCode::PushBool => {
                 // Pushes a boolean value onto the stack.
                 let boolean: bool = instruction.read_operand(0);
-                self.state.push(boolean);
+                self.state_mut().push(boolean);
+                self.state_mut().program_counter += 1;
             }
 
             OpCode::PushNull => {
-                println!("PushNull is no longer valid op code, because null is no longer a valid value from Yarn Spinner 2.0 onwards. To fix this error, re-compile the original source code.");
+                panic!("PushNull is no longer valid op code, because null is no longer a valid value from Yarn Spinner 2.0 onwards. To fix this error, re-compile the original source code.");
             }
             OpCode::JumpIfFalse => {
                 // Jumps to a named label if the value on the top of the stack evaluates to the boolean value 'false'.
-                let is_top_value_true: bool = self.state.pop();
+                let is_top_value_true: bool = self.state_mut().pop();
                 if !is_top_value_true {
                     let label_name: String = instruction.read_operand(0);
-                    let instruction_point = self.find_instruction_point_for_label(&label_name) - 1;
-                    self.state.program_counter = instruction_point;
+                    let instruction_point = self.find_instruction_point_for_label(&label_name);
+                    self.state_mut().program_counter = instruction_point;
+                } else {
+                    self.state_mut().program_counter += 1;
                 }
             }
             OpCode::Pop => {
                 // Pops a value from the stack.
-                self.state.pop_value();
+                self.state_mut().pop_value();
+                self.state_mut().program_counter += 1;
```

Comment:
pop does +=1? Ok. :)

Reply:
we have advanced one step (which we just popped), which makes sense

Reply:
The code did this before as well, it was just more hidden until this refactor.